docs/content/best-practices/github-workflow.md

```diff
@@ -0,0 +1,162 @@
+
+## Setting up the Workflow
+Now that we know the parts that are needed in a workflow, we can learn how to set one up. Here are the steps we will go through to set up the workflow: 
+
+* [Making yaml files](#making-yaml-files)
+* [Setting up secrets in your repository](#setting-up-secrets-in-your-repository)
+* [Using credential files](#using-credential-files)
+
+### Making yaml files
+The way you set up your yaml files depends on who will be contributing to your repository. If you are the only one working on a project and no one else can make pull requests (you have a private repository), you can set up one yaml file to run when a pull request opens and one yaml file to run when a commit is merged. You can have the one that runs on a pull request run your bundle to test it, and you can have the second yaml file publish the bundle to Docker Hub. 
+
+If you are not the only one contributing to the repository and other contributors will be making pull requests from forks, the yaml file setup is slightly different. GitHub actions and workflows do not run automatically on pull requests from a fork. So, in this situation, you will only need one yaml file that runs when a pull request is merged. You can test the bundle and publish in the same yaml file.  
+
+### Setting up secrets in your repository
+In order to publish your bundle to Docker Hub, you will need to set up secrets. You can add these in the settings of your repository and then use them in your workflow.
+
+### Using credential files
+If you were using credentials in your bundle, you will need to set up a credential file in your repository to use with your workflow. After you run `porter credentials generate`, you should see a JSON file with the name of your bundle in the ~/.porter/credentials directory. You need to add this file to your repository so you can pass in the file as the credential. For example, if my bundle was named hello, there would be a file with the credentials called hello.json. I would do the following to install my bundle:
+```yaml
+porter install -c ./hello.json
+```
+
+## Example Code
+
+Now, we will show example code for a workflow and then explain what the code does.
+
+* [Show example code](#show-example-code)
+* [Explain the code](#explain-the-code)
+
+### Show example code
+```yaml
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push event for the master branch
+on:
+  push:
+    branches: [ main ]
+
+# Set up environment variables needed for the bundle
+env:
+  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+# A workflow run
+jobs:
+  publish: 
+    runs-on: ubuntu-latest
+    
+    steps: 
+    # Check out code
+    - uses: actions/checkout@v1
+    # Use Porter GH action to set up Porter
+    - name: Setup Porter
+      uses: deislabs/porter-gh-action@v0.1.1
+      with:
```

Comment:
Since people are going to copy/paste this, and we aren't going to update this example often, I think it's a good idea to not set the porter_version. Otherwise people will end up with old versions of porter accidentally. 

In your intro, we should keep telling people how to set the version of Porter. That's is useful, but the final example perhaps shouldn't be hard coded to what is going to quickly become an old version of Porter.